src/js/media-fullscreen-button.js

```diff
@@ -15,31 +15,28 @@ const exitFullscreenIcon = `
 </svg>
 `;
 
+const api = {
+  enter: "requestFullscreen",
+  exit: "exitFullscreen",
+  event: "fullscreenchange",
+  element: "fullscreenElement",
+  error: "fullscreenerror",
+};
+
+if (!document.fullscreenElement) {
+  api.enter = "webkitRequestFullScreen";
+  api.exit = document.webkitExitFullscreen != null ? "webkitExitFullscreen" : "webkitCancelFullScreen";
+  api.event = "webkitfullscreenchange";
+  api.element = "webkitFullscreenElement";
+  api.error = "webkitfullscreenerror";
+}
+
 class MediaFullscreenButton extends MediaChromeButton {
   constructor() {
     super();
 
     this.icon = enterFullscreenIcon;
-  }
-
-  onClick() {
-    const api = this.getFullscreenApi(this.mediaChrome);
-
-    if (this.mediaChrome == document[api.element]) {
-      api.exit();
-    } else {
-      if (document.pictureInPictureElement) {
-        // Should be async
-        document.exitPictureInPicture();
-      }
-
-      api.enter();
-    }
-  }
 
-  mediaSetCallback() {
-    const api = this.getFullscreenApi(this.mediaChrome);
-    
     document.addEventListener(api.event, () => {
       if (this.mediaChrome == document[api.element]) {
         this.icon = exitFullscreenIcon;
```

Comment:
Now that the api mapping is static, the event listener can be added in the constructor